component/core/src/main/java/org/exoplatform/social/core/utils/MentionUtils.java

```diff
@@ -39,7 +41,17 @@ public static String substituteUsernames(String portalOwner, String message) {
                 continue;
             }
             try {
-                username = LinkProvider.getProfileLink(username, portalOwner);
+                if(activity.getActivityStream().isSpace() && activity.getSpaceId() != null) {
+                    SpaceService spaceService = CommonsUtils.getService(SpaceService.class);
+                    boolean isMember = spaceService.isMember(spaceService.getSpaceById(activity.getSpaceId()),identity.getRemoteId());
+                    if (isMember) {
+                        username = LinkProvider.getProfileLink(username, portalOwner);
+                    } else {
+                        username = identity.getProfile().getFullName();
+                    }
+                } else {
+                    username = LinkProvider.getProfileLink(username, portalOwner);
+                }
```

Comment:
No need to make the check here. In fact, the `MentionUtils` should be generic and doesn't change old activities when a member leaves a space by example.

```suggestion
```